### PR TITLE
Support new fields for /v1/images/generation API

### PR DESCRIPTION
**Pull request by mikecutalo**
**Describe the change**
Updating the [/v1/images/generation](https://platform.openai.com/docs/api-reference/images/create) api with the new changes announced at [devday](https://openai.com/blog/new-models-and-developer-products-announced-at-devday).

**Describe your solution**
Simply add new fields to `ImageRequest` & `ImageResponseDataInner` as well a bunch of new const's for supported options.

**Tests**
Added more fields to the existing `TestImages` test, as these are later unmarshaled. Although if you would like to see tests that simply assert marshalling I could add some, let me know.

Issue: https://github.com/sashabaranov/go-openai/issues/529

### CODE REVIEW

**Comment by vuguet** [comment]
could you  set the response on this way ? : 

```
	type ImageResponseDataInner struct {
		URL     string `json:"url,omitempty"`
		B64JSON string `json:"b64_json,omitempty"`
		RevisedPrompt string `json:"revised_prompt"`
	}
```

They now return a prompt in the response : 
image

**Comment by vuguet** [comment]
It is not related to your changes and sorry for so many comments but the linter issue is because of this : 
image

**Comment by mikecutalo** [comment]
Nice good catch let me update the response object. There is already a lint fix PR https://github.com/sashabaranov/go-openai/pull/522 thats been opened.

File: image.go [hunk 1]
```diff
@@ -13,18 +13,38 @@ const (
...
 	CreateImageResponseFormatB64JSON = "b64_json"
 )
 
+const (
+	CreateImageModelDallE2 = "dall-e-2"
+	CreateImageModelDallE3 = "dall-e-3"
+)
+
+const (
+	CreateImageQualityHD = "hd"
```
Comment (vvatanabe) [inline image.go:32]: @mikecutalo 
Sometimes you may want to explicitly specify default values in the code. Please add the following as well.

`CreateImageQualityStandard = "standard"`

image

https://platform.openai.com/docs/api-reference/images/create#images-create-quality

**Review by vvatanabe — APPROVED** [review]

**Comment by vvatanabe** [comment]
@mikecutalo Thanks for implementing the new API so quickly!

**Pull request merged**